scene/gui/graph_edit.cpp

```diff
@@ -337,6 +337,16 @@ const List<Ref<GraphEdit::Connection>> &GraphEdit::get_connection_list() const {
 	return connections;
 }
 
+int GraphEdit::get_connection_count(const StringName &p_node, int p_port) {
+	int count = 0;
+	for (const Ref<Connection> &conn : connections) {
+		if ((conn->to_node == p_node || conn->from_node == p_node) && (conn->to_port == p_port || conn->from_port == p_port)) {
```

Comment:
I'm not sure if this condition is correct. E.g. if `to_node` is `p_node` and `from_port` is `p_port` (in another node), it could count a connection to another port.

Reply:
This allows us to check the right ports as well as the left ones, without this only the left or right port is counted

Reply:
The test project did not show any problems, this allows us for example to test one node that is attached to itself, from left to right, which is the correct behavior

Reply:
This returns 2 connections to Port 0 in the middle node.
![image](https://github.com/user-attachments/assets/1173516e-947d-4c65-981d-06a3b89d80b4)

Reply:
This is actually happening... I had an idea to create two functions, to check `from_node from_port` and `to_node to_port`, but it seems redundant, what else can I come up with to fix this, any tips?

Reply:
You just have to rearrange your condition, so it always checks the correct port.

Reply:
> You just have to rearrange your condition, so it always checks the correct port, i.e. use `from_port` only if connecting from this node and `to_port` if connecting to this node.

The logic for checking the condition has been changed, now it works as expected